### PR TITLE
OCPCLOUD-1796: Improve test coverage

### PR DESCRIPTION
**Pull request by RadekManak**
This PR improves the test coverage by implementing unit and e2e tests.

Additionaly:
- Adds pytest-cov html coverage page into artifacts directory
- Increases test verbosity to keep full equal-expected diff
- Adds check for valid tag policy
- Solves # DD - Deleted/Deleted conflict

### CODE REVIEW

**Comment by RadekManak** [comment]
Even if it embeds the page, then the report does not work inside the iframe. Having it in artifacts is the best thing we can do.

File: rebasebot/bot.py [hunk 1]
```diff
@@ -466,6 +474,60 @@ def _manual_rebase_pr_in_repo(repo: github3.github.repo.Repository) -> Optional[
     return None
 
 
+def _push_rebase_branch(gitwd, rebase) -> bool:
```
Comment (lobziik) [inline rebasebot/bot.py:477]: Would be cool to have type annotations here. And on all newly introduced functions.

File: tests/test_rebases.py [hunk 2]
```diff
@@ -54,19 +56,22 @@ def test_workdir_init(self, working_repo_context):
         assert len(commits) == 1
         assert commits[0].message == "Upstream commit\n"
 
-        working_repo_dir_content = {i.name for i in os.scandir(working_repo_path)}
+        working_repo_dir_content = {
+            i.name for i in os.scandir(working_repo_path)}
```
Comment (lobziik) [inline tests/test_rebases.py:60]: nit: can we move closing bracket on a newline?

**Comment by JoelSpeed** [comment]
/assign @elmiko 

I think you are best placed to review this one

**Review by elmiko — COMMENTED** [review]
i'm still reviewing some of the tests, but wanted to get these comments shared. apologies for not getting this in sooner @RadekManak

**Comment by RadekManak** [comment]
Changes since last review:
https://github.com/openshift-eng/rebasebot/compare/1c44d5eec397298a81dbb08249db63fe03641575..2463085ffc3364963782d17e81d935e12e1e0a9b

**Comment by elmiko** [comment]
updates are nice, thanks @RadekManak 

/lgtm

**Pull request merged**